### DISCUSSION TITLE
How to set up the transparent proxy

### DISCUSSION THREAD

**Discussion by jdearmas** [mitmproxy/mitmproxy#4904] [Q&A] [answered]
Does the client and the proxy have to be connected via router? 

If it doesn't (point-to-point) how do you configure the client to use the proxy as a gateway? Do you need to setup a DHCP server? Am I over complicating this? 

Thanks.

**Comment by jdearmas** [comment]
RTFM 

https://docs.mitmproxy.org/stable/concepts-howmitmproxyworks/#transparent-http

It clearly says router right there. 😖😤

**Comment by mhils** [comment, marked as answer]
Hi @jdearmas! 😃 

First off, make sure that you actually need transparent mode. Setting up a normal proxy or a reverse proxy is much more straightforward. mitmproxy's regular mode now also redirects based on HTTP Host headers, so just running that + some hardcoded DNS entries also works in many cases. Anyhow, coming back to your transparent mode question:

1. For transparent mode they key parts are on https://docs.mitmproxy.org/stable/concepts-modes/. You need to get the packets to your mitmproxy machine _without_ changing the destination IP address (and then into mitmproxy with some iptables rules). mitmproxy can then ask the OS what the actual destination IP address is.
2. To get the packets to the mitmproxy machine you can do one out of three things: 
  a. Let the target device connect via the mitmproxy machine, for example by hosting a wireless hotspot on there. 
  b. Use a router where you have shell access to perform redirection (both devices connected to this router).
  c. Alternatively, as a hack, you can often set the mitmproxy machine as a custom gateway in the wireless network config.

**Reply by jdearmas** [reply]
Hey @mhils! Thank you for the reply!

My use case is to monitor and intercept traffic leaving the client (traffic generated from a 3rd-party non-browser application). From what I understand, a transparent proxy is the tool for the job ( please tell me otherwise ). 

Your recommendations make sense and I'll try them out. The root issue for me was figuring how to get client packets to go the proxy machine, as you've stated, but purely point-to-point. 

By point-to-point, I mean one ethernet cable between the proxy and client and thats it. The proxy is connected to the internet wirelessly. 

But thanks to your recommendations, I now understand I can connect everything to a router and configure the network from there. 

Thank you for the time and the help!

**Reply by mhils** [reply]
Which OS are you on?

**Reply by jdearmas** [reply]
The client machine is Windows 10. 
The proxy machine is Kali Linux.

**Reply by mhils** [reply]
> From what I understand, a transparent proxy is the tool for the job ( please tell me otherwise ).

Transparent mode definitely is an option here (and the best one if your network setup allows it). If you are interested in seeing the contents a tool like https://www.proxifier.com/ should also work. Alternatively, you can run mitmproxy on Windows with `--mode transparent` and it should pick up traffic from the same machine automatically. That's broken in 7.0.4 unfortunately, but I've just committed a fix that should be available at https://mitmproxy.org/downloads/#branches/main/ in a few minutes. :)

**Reply by jdearmas** [reply]
Hey @mhils!

I used the latest version on my windows machine yesterday. You were right, I didnt even need have another machine. So thanks for that!
